FutureWorkshops/FW-Android-Playground
# Chore: Refactor marvel-module to introduce Kotlin - pt 1

### :tophat: What is the goal?

To start introducing Kotlin in our FW architecture project, starting from the `marvelHeroes` module. In order for this to be reviewable, I intend to do this in 3 parts. For the first part, I focused on the Network and Data layers. In the next PR I will refactor the rest of the `domain` package and change the way Dagger is used, to be more inline with Kotlin guidelines. In the third I will refactor the View layer

### How is it being implemented?

By configuring Kotlin in the module and progressively refactor the various layers of the app from Java to Kotlin. 

### How can it be tested?

_Pull the branch, build it and run it locally to verify the project behaves correctly._

### Code Documentation Wiki 

- Upgraded gradle plugin from 3.0.1 to 3.1.2
- Configured Kotlin in the marvel module (v.1.2.41)
- Changed annotation processor to KAPT
